openmdao/core/problem.py

```diff
@@ -1890,40 +1936,65 @@ def _assemble_derivative_data(derivative_data, rel_error_tol, abs_error_tol, out
+                            num_bad_jacs += 1
                         if out_stream:
-                            out_stream.write('    Relative Error {}: {}'.format(desc, error_str) +
+                            out_buffer.write('    Relative Error {}: {}'.format(desc, error_str) +
                                              '\n')
+
                     if out_stream:
-                        out_stream.write('\n')
+                        out_buffer.write('\n')
 
                     # Raw Derivatives
                     if out_stream:
-                        out_stream.write('    Raw Forward Derivative (Jfor)\n')
-                        out_stream.write(str(forward) + '\n')
-                        out_stream.write('\n')
+                        out_buffer.write('    Raw Forward Derivative (Jfor)\n')
+                        out_buffer.write(str(forward) + '\n')
+                        out_buffer.write('\n')
 
                     if not totals and system.matrix_free:
                         if out_stream:
-                            out_stream.write('    Raw Reverse Derivative (Jfor)\n')
-                            out_stream.write(str(reverse) + '\n')
-                            out_stream.write('\n')
+                            out_buffer.write('    Raw Reverse Derivative (Jfor)\n')
+                            out_buffer.write(str(reverse) + '\n')
+                            out_buffer.write('\n')
 
                     if out_stream:
-                        out_stream.write('    Raw FD Derivative (Jfd)\n')
-                        out_stream.write(str(fd) + '\n')
-                        out_stream.write('\n')
+                        out_buffer.write('    Raw FD Derivative (Jfd)\n')
+                        out_buffer.write(str(fd) + '\n')
+                        out_buffer.write('\n')
 
                     if out_stream:
-                        out_stream.write(' -' * 30 + '\n')
+                        out_buffer.write(' -' * 30 + '\n')
+
+                # End of if compact print if/else
+            # End of if not suppress_output
+        # End of for of, wrt in sorted_keys
+    # End of for system in system_list
+
+        if not show_only_incorrect or num_bad_jacs:
+            if out_stream and not suppress_output:
+                out_stream.write(out_buffer.getvalue())
+
+    if not suppress_output and compact_print and not totals:
+        if worst_subjac:
+            worst_subjac_header = \
+                "Sub Jacobian with Largest Relative Error: {1} '{2}'".format(*worst_subjac)
+            out_stream.write('\n' + '#' * len(worst_subjac_header) + '\n')
+            out_stream.write("{}\n".format(worst_subjac_header))
+            out_stream.write('#' * len(worst_subjac_header) + '\n')
+            out_stream.write(header + '\n')
+            out_stream.write('-' * len(header) + '\n')
+            out_stream.write(worst_subjac_line + '\n')
 
 
 def _format_if_not_matrix_free(matrix_free, val):
```

Comment:
I don't think this `quotes` argument is ever used.

Reply:
It is here, right? https://github.com/hschilling/OpenMDAO/blob/6d40866d9bed2fd1c2374cc9c6d7444342d6fe74/openmdao/core/problem.py#L1748

Reply:
as a positional argument.

Reply:
Oh, I thought pep (or good practice) was that you should always use keyword arguments with the keyword specified, rather than as a positional.

Reply:
probably ! Will fix